### PR TITLE
Add Documentation for secnetperf

### LINKED ISSUE #2217: How to test performance
### Discussed in https://github.com/microsoft/msquic/discussions/2216

<div type='discussions-op-text'>

<sup>Originally posted by **Zhangyida183** December  9, 2021</sup>
How to test msquic performance? Is it performance.ps1? What are the parameters ? Can msquic start a client and server to test performance ？

I just used cmake build msquic, and test quicsample(server and client).
Now I want to test msquic performance on my centos, but didn't know how to test it.
</div>

### PR DESCRIPTION
**Pull request by nibanks**
## Description

Closes #2217.

Adds new documentation readme for secnetperf.

### CODE REVIEW

**Review by mtfriesen — APPROVED** [review]

File: src/perf/readme.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+> secnetperf.exe -test:tput -target:127.0.0.1 -upload:10000 -timed:1 -stats:1
+```
+
+### Required
+
+Argument | Usage | Meaning
+--- | --- | ---
+target,server | `-target:<hostname_or_IP>` | The target server to connect to. May be a hostname or an IP address.
+download | `-download:<####>` | The length of data (or time with `-timed:1` arg) to receive. Mutually exclusive with `-upload` arg.
+upload | `-upload:<####>` | The length of data (or time with `-timed:1` arg) to send. Mutually exclusive with `-download` arg.
```
Comment (csujedihy) [inline src/perf/readme.md:60]: The length of data in bytes

File: src/perf/readme.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ip | `-ip:<0,4,6> ` | A hint for the resolving the hostname to an IP address.
+cibir | `-cibir:<hex_bytes>` | The well-known CIBIR identifier.
+encrypt | `-encrypt:<0,1>` | Enables/disables encryption.
+sendbuf | `-sendbuf:<0,1>` | Whether to use send buffering.
+pacing | `-pacing:<0,1>` | Whether to use pacing.
+timed | `-timed:<0,1>` | Indicates the `upload` & `download` arg represent time (ms).
+iosize | `-iosize:<####>` | The size of each send request queued.
+tcp | `-tcp:<0,1>` | Indicates TCP/TLS should be used instead of QUIC. **Windows Only**
+stats | `-stats:<0,1>` | Prints out statistics at the end of each connection.
+sstats | `-sstats:<0,1>` | Prints out statistics at the end of each stream.
```
Comment (csujedihy) [inline src/perf/readme.md:80]: Prints out stream level statistics at the end of each stream?

The current wording sounds like `stats` and `sstats` just print the same statistics on different timings.

File: src/perf/readme.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+cipher | `-cipher:<value>` | Decimal value of 1 or more `QUIC_ALLOWED_CIPHER_SUITE_FLAGS`.
+cpu | `-cpu:<cpu_indexes>` | Comma-separated list of CPUs to run on.
+ecn | `-ecn:<0,1>` | Enables sender-side ECN support.
+exec | `-exec:<lowlat,maxtput,scavenger,realtime>` | The execution profile used for the application.
+pollidle | `-pollidle:<time_us>` | The time, in microseconds, to poll while idle before sleeping (falling back to interrupt-driven IO).
+stats | `-stats:<0,1>` | Prints out statistics at the end of each connection.
+
+# Client
+
+Since tests are client, the client side of secnetperf generally has several arguments passed in. Many of the arguments are also test specific, but the following (overlap with server) do apply to all:
```
Comment (csujedihy) [inline src/perf/readme.md:29]: "Since tests are client" I don't quite understand this

File: src/perf/lib/ThroughputClient.cpp [hunk 4]
```diff
@@ -26,8 +26,8 @@ PrintHelp(
         "  -target:<####>               The target server to connect to.\n"
 #if _WIN32
         "  -comp:<####>                 The compartment ID to run in.\n"
-        "  -core:<####>                 The CPU core to use for the main thread.\n"
 #endif
+        "  -core:<####>                 The CPU core to use for the main thread.\n"
```
Comment (csujedihy) [inline src/perf/lib/ThroughputClient.cpp:30]: The CPU cores?

File: src/perf/lib/ThroughputClient.cpp
Comment (hero-1-hero) [inline src/perf/lib/ThroughputClient.cpp:1]: Hi, I saw the - qtip:<0/1>parameter in SecNetPerfMain. Setting this parameter will result in a 100% CPU utilization rate. I would like to give you feedback, and there is also no introduction to this parameter in the readme document.

**Comment by hero-1-hero** [comment]
Hi, I saw the - qtip:<0/1>parameter in SecNetPerfMain. Setting this parameter will result in a 100% CPU utilization rate. I would like to give you feedback, and there is also no introduction to this parameter in the readme.me document.

**Pull request merged**